fog/view/load-test/src/main.rs

```
@@ -41,79 +35,173 @@ struct Config {
+    }
+}
+
+impl core::ops::Sub for Counters {
+    type Output = Self;
+    fn sub(self, other: Self) -> Self {
+        Self {
+            num_requests: self.num_requests - other.num_requests,
+            num_errors: self.num_errors - other.num_errors,
+            total_millis_latency: self.total_millis_latency - other.total_millis_latency,
+        }
+    }
 }
 
 fn worker_thread(
     uri: String,
     grpc_retry_config: GrpcRetryConfig,
-    account_key: AccountKey,
     num_search_keys: usize,
-    num_reqs: Arc<AtomicU64>,
+    counters: Arc<Mutex<Counters>>,
+    stop_requested: Arc<AtomicBool>,
     logger: Logger,
 ) {
     let mut fog_view_client = build_fog_view_conn(&uri, grpc_retry_config, &logger);
 
     let resp = fog_view_client
         .request(0, 0, Default::default())
+        .map_err(|err| {
+            stop_requested.store(true, Ordering::SeqCst);
+            err
+        })
         .expect("request");
     let rng_record = &(resp.rng_records[0]);
-    let rng = VersionedKexRng::try_from_kex_pubkey(
-        &rng_record.pubkey,
-        &account_key.default_subaddress_view_private(),
-    )
-    .expect("kex");
+    let private_key = FromRandom::from_random(&mut rand::thread_rng());
+    let rng = VersionedKexRng::try_from_kex_pubkey(&rng_record.pubkey, &private_key).expect("kex");
     let search_keys = rng.take(num_search_keys).collect::<Vec<Vec<u8>>>();
 
-    loop {
-        let _resp = fog_view_client
-            .request(0, 0, search_keys.clone())
-            .expect("request");
-        num_reqs.fetch_add(1, Ordering::SeqCst);
+    while !stop_requested.load(Ordering::SeqCst) {
+        let start = Instant::now();
+        let result = fog_view_client.request(0, 0, search_keys.clone());
+        let duration = Instant::now().duration_since(start);
+
+        let mut counters = counters.lock().unwrap();
+        counters.num_requests += 1;
+        counters.total_millis_latency += u64::try_from(duration.as_millis()).unwrap();
```

why not `duration.as_millis() as u64`? did clippy complain?

---

i think it's subtly different, i don't believe that `duration.as_millis() as u64` will panic if there is u64 overflow, i think it just silently truncates, which could lead to weird data. but even so, it's not going to overflow. i'm happy to do it that way also.

---

FYI u64 milliseconds > 580million years, https://www.google.com/search?q=(2^64-1)%2F(1000*60*60*24*365)

---

good point about the `+=`. i think i'm just going to leave it as is